### PR TITLE
Concatenation for sequence types.

### PR DESCRIPTION
**Pull request by dcjones**
This adds a concatenation constructor for sequence types, like

```julia
julia> DNASequence(dna"ACGTACGT", dna"TTTTTTTT", dna"TCGATCGA")
24nt DNA Sequence
 ACGTACGTTTTTTTTTTCGATCGA
```

### CODE REVIEW

**Comment by bicycle1885** [comment]
How about defining `(*)` and `(^)` operators like string types?

**Comment by dcjones** [comment]
Now with `*` and `^`. :+1:

**Comment by bicycle1885** [comment]
Neat!

**Comment by brk00** [comment]
Neat, nice work! But I believe there is a typo on tests file (repitition -> repetition)

**Comment by dcjones** [comment]
Thanks @brk00, at least I'm consistent in my misspellings. :wink:

This is ready to merge if cool with @Ward9250 and @Blahah.

File: src/seq/nucleotide.jl [hunk 1]
```diff
@@ -181,6 +181,124 @@ function NucleotideSequence{T<:Nucleotide}(::Type{T}, seq::String)
...
+
+This is unsafe in the following ways:
+    * Disregards immutability of `dest`
+    * May write a few bases past dest[pos + length(src) - 1]
+    * Doesn't bounds check anything.
+
+It's really only suitable for use in the concatenation constructor.
+
+""" ->
+function Base.unsafe_copy!{T}(dest::NucleotideSequence{T}, pos::Int, src::NucleotideSequence{T})
```
Comment (brk00) [inline src/seq/nucleotide.jl]: Why not imorting this functio before extending it?

**Comment by brk00** [comment]
Excited about this PR. Once it get merged I can start documenting Seq module.

**Comment by dcjones** [comment]
I'll merge this a little later today unless there are explicit objections. It's a self-contained obvious feature with 100% test coverage. Should be uncontroversial.

**Comment by Ward9250** [comment]
No objections from me!

**Comment by Blahah** [comment]
Looks good to me.

**Comment by dcjones** [comment]
Ok, thanks!

**Pull request merged**